eppo_core/src/events/vec_event_queue.rs

```diff
@@ -14,15 +16,22 @@ pub trait EventQueue {
     /// Returns whether the queue contains enough Pending events for delivering *at least* one batch.
     fn is_batch_full(&self) -> bool;
 
+    /// Marks the provided events as failed and increments their attempts and adds it to the failed queue.
     fn mark_events_as_failed(&self, failed_events: Vec<QueuedEvent>);
 }
 
+#[derive(Debug, Clone)]
+pub struct VecEventQueueConfig {
+    pub batch_size: usize,
+    pub max_queue_size: usize,
+    pub max_retries: u8,
+}
+
 /// A simple event queue that stores events in a vector
 #[derive(Debug, Clone)]
 pub struct VecEventQueue {
-    batch_size: usize,
-    max_queue_size: usize,
-    event_queue: Arc<Mutex<HashMap<QueuedEventStatus, VecDeque<QueuedEvent>>>>,
+    config: VecEventQueueConfig,
+    event_queue: Arc<Mutex<HashMap<QueuedEventStatus, LinkedHashSet<QueuedEvent>>>>,
```

Comment:
nit: hash map on event status is somewhat inefficient given there's just two statuses that we're interested in. Two fields would avoid the hashmap lookup + split the mutex potentially decreasing contention? Pending queue could also continue using `VecDeque`, which is more efficient than `LinkedHashSet`

Reply:
we don't want duplicate events, thus the `LinkedHashSet`. I agree on using a map being somewhat pointless, I can split into separate queues